### PR TITLE
crfsuite 0.12 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by dawidd6 — DISMISSED** [review]

**Review by chenrui333 — DISMISSED** [review]

File: Formula/crfsuite.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class Crfsuite < Formula
+  desc "Fast implementation of conditional random fields"
+  homepage "https://www.chokkan.org/software/crfsuite/"
+  url "https://github.com/downloads/chokkan/crfsuite/crfsuite-0.12.tar.gz"
```
Comment (bayandin) [inline Formula/crfsuite.rb:4]: Wow, I didn't know there's such links like `https://github.com/downloads/...`

I'd suggest switching it to the format we usually use:

```suggestion
  url "https://github.com/chokkan/crfsuite/archive/refs/tags/0.12.tar.gz"
```

File: Formula/crfsuite.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  sha256 "e7fc2d88353b1f4de799245f777d90f3c89a9d9744ba9fbde8c7553fa78a1ea1"
+  license "BSD-3-Clause"
+
+  depends_on "python@3.11" => :test
```
Comment (bayandin) [inline Formula/crfsuite.rb:8]: Do we build python bindings for it?
Should we add `:build` for `python@3.11` as well? Or maybe switching to `uses_from_macos "python"`?

File: Formula/crfsuite.rb (hunk 1 above)
Comment (alebcay) [reply Formula/crfsuite.rb:4]: Good catch. This is borrowing from an existing formula in the `homebrew-science` (legacy) tap.

File: Formula/crfsuite.rb (hunk 2 above)
Comment (alebcay) [reply Formula/crfsuite.rb:8]: Switching to `uses_from_macos "python"`. This package does not build Python bindings. An example is provided in the repo which uses a Python script to convert the data set resource that we download into a `crfsuite`-compatible format. This Python script is not really considered as part of the application (or, at least `make install` doesn't install it).

File: Formula/crfsuite.rb (hunk 1 above)
Comment (alebcay) [reply Formula/crfsuite.rb:4]: Actually, these two tarballs don't have the same contents. The original one I had includes a pre-generated `configure` file. The `/archive` tarball doesn't have it.

It seems there are other formulae in core tap that use this kind of link, e.g. `apktool`, `jadx`.

**Review by chenrui333 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

File: Formula/crfsuite.rb (hunk 1 above)
Comment (bayandin) [reply Formula/crfsuite.rb:4]: Oh, wow, old stuff!
TIL there was a downloads API that was replaced with releases like 10 years ago.

Source: https://stackoverflow.com/questions/63292793/how-can-i-upload-files-to-https-github-com-downloads-user-repo

**Pull request closed without merging**